packages/typespec-azure-resource-manager/lib/common-types/types-ref.tsp

```diff
@@ -233,6 +295,17 @@ namespace Azure.ResourceManager.CommonTypes;
   Azure.ResourceManager.CommonTypes.Versions.v5
 );
 
+@@resourceParameterBaseFor(ApiVersionParameter.apiVersion,
```

Comment:
This should not impact autorest serialization, why move this here?

Reply:
I am moving these here to keep the type.tsp pure from decorations as later want to transfer ownership to ARM team.